pkg/controller/machine/drain_controller.go

```diff
@@ -182,3 +191,41 @@ func (d *machineDrainController) drainNode(ctx context.Context, machine *machine
 
 	return nil
 }
+
+// isDrainAllowed checks whether the drain is permitted at this time.
+// It checks the following:
+// - Is the node cordoned, if so allow draining to complete any previous attempt to drain.
+// - Is the node a control plane node, if so, only allow draining if no other control plane node is already being drained.
+func (d *machineDrainController) isDrainAllowed(ctx context.Context, node *corev1.Node) error {
+	if node.Spec.Unschedulable {
+		// If the node has already been cordoned, continue to drain.
+		return nil
+	}
+
+	if !isControlPlaneNode(*node) {
+		// We always allow draining of worker nodes.
+		return nil
+	}
+
+	nodes := &corev1.NodeList{}
+	if err := d.Client.List(ctx, nodes); err != nil {
+		return fmt.Errorf("could not list control plane nodes: %v", err)
+	}
+
+	for _, otherNode := range nodes.Items {
+		if isControlPlaneNode(otherNode) && otherNode.Spec.Unschedulable {
+			klog.Warningf("Drain not permitted for node %q: found other control plane node already draining %q, do not continue until the other node is removed", node.Name, otherNode.Name)
```

Comment:
I suggest to rephrase this message and outline that another node is "cordoned and presumably in a middle of the drain procedure". We can not say that another node is already draining, it's an assumption as far as i can say.

Reply:
Updated the wording, WDYT?

Reply:
lgtm, thanks